src/types.ts

```diff
@@ -24,3 +24,9 @@ export type UsePromise<T, Args extends any[], E = string> = [
     PromiseLoader<void, Args>,
     () => void,
 ];
+export type UsePromiseWithInterval<T, E, A extends any[]> = [
```

Comment:
Maybe let's use Typescript's named tuples here, it'd improve developer experience in IDEs - https://www.typescriptlang.org/play?ts=4.0.2#example/named-tuples
Not sure if the `UsePromise` type uses it, if not let's add it there as well.

Reply:
Done, pls check